packages/tupaia-web/src/pages/ModalRoutes.tsx

```diff
@@ -40,8 +47,12 @@ export const ModalRoutes = () => {
   if (!modal || !Object.values(MODAL_ROUTES).includes(modal)) return null;
 
   const ModalView = modalViews[modal];
+
+  const onCloseModal = () => {
+    closeModal(modalParams[modal as keyof typeof modalParams]);
```

Comment:
Ah I see you are clearing the params onclose even if the form is not submitted. This actually makes quite a lot of sense now that I see it in action.